File: app/src/main/java/com/jakebarnby/filemanager/sources/SourceListener.java
package com.jakebarnby.filemanager.sources;

import com.jakebarnby.filemanager.sources.models.SourceFile;
import com.jakebarnby.filemanager.util.TreeNode;

public interface SourceListener {
    void onCheckPermissions(String permission, int requestCode);
    void onNoConnection();
    void onLoadStarted();
    void onLoadAborted();
    void onLoadError(String errorMessage);
    void onLoadComplete(TreeNode<SourceFile> rootFile);
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/SourceListener.kt
package com.jakebarnby.filemanager.sources

import com.jakebarnby.filemanager.sources.models.SourceFile
import com.jakebarnby.filemanager.util.TreeNode

interface SourceListener {
    fun onCheckPermissions(permission: String, requestCode: Int)
    fun onNoConnection()
    fun onLoadStarted()
    fun onLoadAborted()
    fun onLoadError(errorMessage: String?)
    fun onLoadComplete(rootFile: TreeNode<SourceFile>)
    fun onLogout()
}